clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -2217,7 +2275,9 @@ public void processConsumerRecord(
       if (needToUnsub) {
         errorMessage += ". Consumption will be halted.";
         statusReportAdapter.reportError(Collections.singletonList(partitionConsumptionState), errorMessage, e);
-        unSubscribePartition(new PubSubTopicPartitionImpl(versionTopic, faultyPartition));
+        // As unsubscription needs to wait for all records drilled and then proceed, wait here will cause endless wait.
+        Runnable runnable = () -> unSubscribePartition(new PubSubTopicPartitionImpl(versionTopic, faultyPartition));
+        CompletableFuture.runAsync(runnable);
```

Comment:
Endless wait is not good, but if we run it async, it has a chance that the task could be run very late, maybe even after the ingestion task is killed. I don't know if that could cause some issues

Reply:
It's great that you take this part into consideration and did a thorough check! Just to double check my understanding:
Previously the unsubscribe is also done asynchronously by the SIT thread; with the current change, if the new feature is on, you are concerned that the drainer thread will instead do the unsubscribe work for SIT and slow down drainer? If so, can we only use the common pool in Executor when the new feature flag is on?